### PR TITLE
Turn off safeseh for foundation on X86.

### PR DESCRIPTION
**Pull request by hjyamauchi**

### CODE REVIEW

File: build.ps1 [hunk 1]
```diff
@@ -429,6 +429,10 @@ function Build-CMakeProject
     $SwiftArgs.Add("-Xlinker /INCREMENTAL:NO") | Out-Null
     $SwiftArgs.Add("-Xlinker /DEBUG:DWARF") | Out-Null
 
+    # Turn off safeseh for lld as it has safeseh enabled by default
+    # and fails with an ICU data object file icudt69l_dat.obj.
+    $SwiftArgs.Add("-Xlinker /SAFESEH:NO") | Out-Null
```
Comment (compnerd) [inline build.ps1:434]: I don't think that we should disable Safe SEH globally, it has some benefits for attack mitigation.

Comment (hjyamauchi) [reply build.ps1:434]: If it was disabled before the lld switch, not much was lost? Is there a way to disable it selectively?

Comment (hjyamauchi) [reply build.ps1:434]: Maybe I misunderstood "globally". We could disable safe seh in the `Build-Foundation` case only.

Comment (compnerd) [reply build.ps1:434]: Yeah, I think that we should be able to do that.  We can add `-D CMAKE_SHARED_LINKER_FLAGS="/SAFESEH:NO"` to `Build-Foundation`.

Comment (hjyamauchi) [reply build.ps1]: I saw that `CMAKE_SHARED_LINKER_FLAGS` had no effect on the swift linking. Updated it so it passes the option through a parameter instead.

File: build.ps1 [hunk 2]
```diff
@@ -357,7 +357,8 @@ function Build-CMakeProject
     [string[]] $UseBuiltCompilers = @(), # ASM,C,CXX,Swift
     [string] $SwiftSDK = "",
     [hashtable] $Defines = @{},
-    [string[]] $BuildTargets = @()
+    [string[]] $BuildTargets = @(),
+    [string] $ExtraSwiftArgs = ""
```
Comment (compnerd) [inline build.ps1:361]: This is unnecessary

File: build.ps1 [hunk 3]
```diff
@@ -433,6 +434,8 @@ function Build-CMakeProject
     $SwiftArgs.Add("-Xlinker /OPT:REF") | Out-Null
     $SwiftArgs.Add("-Xlinker /OPT:ICF") | Out-Null
 
+    $SwiftArgs.Add($ExtraSwiftArgs) | Out-Null
```
Comment (compnerd) [inline build.ps1:437]: This is unnecessary

File: build.ps1 [hunk 4]
```diff
@@ -823,6 +826,11 @@ function Build-Foundation($Arch, [switch]$Test = $false)
       $Targets = @("default", "install")
     }
 
+    # Turn off safeseh for lld as it has safeseh enabled by default
+    # and fails with an ICU data object file icudt69l_dat.obj. This
+    # matters to X86 only.
+    $ExtraSwiftArgs = "-Xlinker /SAFESEH:NO"
```
Comment (compnerd) [inline build.ps1:832]: We should conditionalise it to x86 and pass it as `CMAKE_Swift_FLAGS`.

**Review by compnerd — CHANGES_REQUESTED** [review]

File: build.ps1 (hunk 4 above)
Comment (hjyamauchi) [reply build.ps1]: Done

File: build.ps1 [hunk 5]
```diff
@@ -823,6 +823,15 @@ function Build-Foundation($Arch, [switch]$Test = $false)
       $Targets = @("default", "install")
     }
 
+    $SwiftFlags = ""
+    if ($Arch -eq $ArchX86)
+    {
```
Comment (compnerd) [inline build.ps1:828]:
```suggestion
    if ($Arch -eq $ArchX86) {
```

Everything is now uniformly K&R.

**Review by compnerd — APPROVED** [review]

File: build.ps1 [hunk 6]
```diff
@@ -823,6 +823,14 @@ function Build-Foundation($Arch, [switch]$Test = $false)
       $Targets = @("default", "install")
     }
 
+    $SwiftFlags = ""
+    if ($Arch -eq $ArchX86) {
+      # Turn off safeseh for lld as it has safeseh enabled by default
+      # and fails with an ICU data object file icudt69l_dat.obj. This
+      # matters to X86 only.
+      $SwiftFlags = "-Xlinker /SAFESEH:NO"
+    }
```
Comment (compnerd) [inline build.ps1:832]:
```suggestion
    # Turn off safeseh for lld as it has safeseh enabled by default
    # and fails with an ICU data object file icudt69l_dat.obj. This
    # matters to X86 only.
    $SwiftFlags = if ($Arch -eq $ArchX86) { "-Xlinker /SAFESEH:NO" } else { "" }
```

File: build.ps1 [hunk 7]
```diff
@@ -834,6 +839,7 @@ function Build-Foundation($Arch, [switch]$Test = $false)
         CMAKE_INSTALL_PREFIX = "$($Arch.SDKInstallRoot)\usr";
         CMAKE_SYSTEM_NAME = "Windows";
         CMAKE_SYSTEM_PROCESSOR = $Arch.CMakeName;
+        CMAKE_Swift_FLAGS = $SwiftFlags;
```
Comment (compnerd) [inline build.ps1:842]:
```suggestion
        # Turn off safeseh for lld as it has safeseh enabled by default
        # and fails with an ICU data object file icudt69l_dat.obj. This
        # matters to X86 only.
        CMAKE_Swift_FLAGS = if ($Arch -eq $ArchX86) { "-Xlinker /SAFESEH:NO" } else { "" };
```

File: build.ps1 [hunk 8]
```diff
@@ -823,6 +823,11 @@ function Build-Foundation($Arch, [switch]$Test = $false)
       $Targets = @("default", "install")
     }
 
+    # Turn off safeseh for lld as it has safeseh enabled by default
+    # and fails with an ICU data object file icudt69l_dat.obj. This
+    # matters to X86 only.
+    $SwiftFlags = if ($Arch -eq $ArchX86) { "-Xlinker /SAFESEH:NO" } else { "" }
+
```
Comment (compnerd) [inline build.ps1:830]:
```suggestion
```

**Pull request merged**